### PR TITLE
Presentation Layer for NPM Register

### PR DESCRIPTION
**Pull request by dgautsch**
This is my first pass at a UI.

Leveraged the Material UI react library to build this out. https://material-ui-1dab0.firebaseapp.com

**Features included**

- Listing of all apps in the registry
- Downloadable Tarballs
- Readme details that open in a modal

**To Do**
- [ ] Add search
- [ ] Unit tests for new API
- [x] S3 tarball download bug fix

### CODE REVIEW

**Comment by dgautsch** [comment]
Open to initial feedback on library choices and code. Any thoughts @jdxcode ?

**Comment by dgautsch** [comment]
Noticed an issue with downloading tarballs on Heroku. Need to implement a fix.

**Comment by jdxcode** [comment]
sorry about the slowness @dgautsch I've been on the road for a couple weeks, but I'm hoping to get some time to go through soon. I've taken a cursory look and it looks great though

File: lib/start.js [hunk 1]
```diff
@@ -2,5 +2,5 @@ const app = require('./server')
 const config = require('./config')
 
 app.listen(config.port, function () {
-  console.log(`${app.name} listening on port ${config.port} [${app.get('env')}]`)
+  console.error(`${app.name} listening on http://localhost:${config.port} [${app.get('env')}]`)
```
Comment (dgautsch) [inline lib/start.js]: This should be log...oversight on my part.

File: lib/storage/base.js [hunk 2]
```diff
@@ -4,6 +4,23 @@ class Base {
   getJSON (key) {
     return this.get(key).then(raw => raw ? JSON.parse(raw) : undefined)
   }
+  createPackageDetails (json) {
+    if (json.name === '' || json.name === undefined) {
```
Comment (dgautsch) [inline lib/storage/base.js]: Probably should refactor this to `if (!json.name)`

**Comment by dgautsch** [comment]
I'll be revisiting this since I have time now. Sorry for the delay everyone.

**Comment by dgautsch** [comment]
@holms yea i'm jumping back into this. stay tuned!

**Comment by holms** [comment]
I'm really glad you've did so much regarding this, would be nice to have some voice from maintainer. Also those CI tests are dead for a long time too. I can't even use this product in container got permission denied if using local fs.

**Comment by holms** [comment]
@dgautsch I've just submitted a PR, and seems to be tests are passing. Probably you're still on old pipeline task or something. I'd re-submit PR if I were you :) Because it's complains about AWS keys, which nowhere to be found in pipeline which been executed on my PR

**Comment by dgautsch** [comment]
Update on this: The only thing holding this back is the tarball from S3 isn't resolving. Works fine when running from the local filesystem. This only happens with namespaced packages. @jdxcode any idea what might be causing this? I was going to look more into this today.

**Comment by holms** [comment]
@jdxcode we want to progress with your beautiful registry :)

**Comment by holms** [comment]
Really waiting for this to be merged :)

**Pull request merged**